src/API/AbstractAPI.php

```diff
@@ -70,11 +72,22 @@ protected function get(string $path, array $query = [])
      */
     protected function post(string $path, array $parameters = [])
     {
-        $response = $this->connection->getHttpClient()->post(
-            $path,
+        $response = $this->client->getHttpClient()->post(
+            $this->buildUrl('post', $path),
             ['json' => $parameters]
         );
 
         return json_decode($response->getBody()->getContents(), true);
     }
+
+    private function buildUrl(string $method, string $path): string
+    {
+        if ($method === 'post') {
+            $baseUri = Arr::get($this->client->getHosts(), 'transactions', $this->client->getHosts()['api']);
+        } else {
+            $baseUri = $this->client->getHosts()['api'];
+        }
```

Comment:
was more thinking along the lines of this @alfonsobries 

```suggestion
    private function buildUrl(string $method, string $path, string $api = 'api'): string
    {
    	$baseUri = $this->client->getHosts()[$api];
```

that way it will default to `api`, but some requests (like the transaction one) can pass their respective api type to use to construct the URL. That way it remains extensible (not limited to an internal check that needs updating) and will prepare it for the evm options as well.

Probably best to not pass it as a string but create an enum for it so the options are limited, and have a failsafe where it defaults to the `"api"` option if the entry doesn't exist in the config?

Reply:
@ItsANameToo my bad at the moment though your comment was on the constructor

What do you think about [this alternative](https://github.com/ArkEcosystem/php-client/pull/111/commits/a9d830c969d406accaca6ab90f7bf1568348f30f), think is a good idea to pass the api o[n the same method that makes the api call](https://github.com/ArkEcosystem/php-client/pull/111/files#diff-1a83adbeff4a8e2977392e6692347cf02a47a1bbba6cdec0fb9072d445b6f182L44-R44) so its clear that it uses a different api

Reply:
@alfonsobries looks good 👍